api_app/serializers.py

```diff
@@ -77,18 +80,194 @@ class _AbstractJobCreateSerializer(rfs.ModelSerializer):
+                    warnings.append(str(e))
+            else:
+                cleaned_analyzer_list.append(a_name)
+
+        return cleaned_analyzer_list
+
+    @staticmethod
+    def filter_connectors(serialized_data: Dict, warnings: List[str]) -> List[str]:
+        # init empty list
+        cleaned_connectors_list = []
+        selected_connectors = []
+
+        # get values from serializer
+        connectors_requested = serialized_data.get("connectors_requested", [])
+        tlp = serialized_data.get("tlp", TLP.WHITE).upper()
+
+        # read config
+        connector_dataclasses = ConnectorConfig.all()
+        all_connector_names = list(connector_dataclasses.keys())
+
+        # run all connectors ?
+        run_all = len(connectors_requested) == 0
+        if run_all:
+            # select all
+            selected_connectors.extend(all_connector_names)
+        else:
+            # select the ones requested
+            selected_connectors.extend(connectors_requested)
+
+        for c_name in selected_connectors:
+            try:
+                cc = connector_dataclasses.get(c_name, None)
+
+                if not cc:
+                    if not run_all:
+                        raise NotRunnableConnector(
+                            f"{c_name} won't run: not available in configuration"
+                        )
+                    # don't add warning if run_all
+                    continue
+
+                if not cc.is_ready_to_use:  # check configured/disabled
+                    raise NotRunnableConnector(
+                        f"{c_name} won't run: is disabled or unconfigured"
+                    )
+
+                if TLP.get_priority(tlp) > TLP.get_priority(
+                    cc.maximum_tlp
+                ):  # check if job's tlp allows running
+                    # e.g. if connector_tlp is GREEN(1),
+                    # run for job_tlp WHITE(0) & GREEN(1) only
+                    raise NotRunnableConnector(
+                        f"{c_name} won't run: "
+                        f"job.tlp ('{tlp}') > maximum_tlp ('{cc.maximum_tlp}')"
+                    )
+            except NotRunnableConnector as e:
+                if run_all:
+                    # in this case, they are not warnings but
+                    # expected and wanted behavior
+                    logger.debug(e)
+                else:
+                    logger.warning(e)
+                    warnings.append(str(e))
+            else:
+                cleaned_connectors_list.append(c_name)
+
+        return cleaned_connectors_list
+
+    def filter_analyzers_and_connectors(self, attrs: dict) -> dict:
+        warnings = []
+        attrs["analyzers_to_execute"] = self.filter_analyzers(
+            attrs,
+            warnings,
+        )
 
+        if not attrs["analyzers_to_execute"]:
+            raise ValidationError(
+                {"detail": "No Analyzers can be run after filtering."}
+            )
+
```

Comment:
Can't we raise the error in `filter_analyzers` directly?

Reply:
Done